### ISSUE TITLE
Improve database structure

### ISSUE THREAD

**Issue by philipwhiuk** [k9mail/k-9#2713]
Raising from #769 comments

### Expected behavior
Database structure should be documented, logical, split into small, easily maintainable tables.

### Actual behavior
Database structure is hard to follow, hard to maintain and fragile. Classes like the MessageListFragment do a single query to retrieve all their data and it's hard to know how well this is tested (probably badly

### Steps to fix

1. Document current database structure.
2. Test MessageListFragment cursor API and other similar core classes, extracting code where necessary
3. Write improved database structure
4. Write migration step

**Comment by philipwhiuk** [comment]
Documenting the current structure shouldn't be too hard. I'll take that and then we can see where we are.

**Comment by Trogel** [comment]
@philipwhiuk Thanks for starting this! If there's anything I can possibly assist with, feel free to let me know.

**Comment by philipwhiuk** [comment]
Thanks for the comment @Trogel - I corrected the typo.
